### PR TITLE
Added Sparkfun ESP32-S3 Thing Plus board

### PR DESCRIPTION
**Pull request by Vigeant**
## Description of Change
Added support for Sparkfun ESP32-S3 Thing Plus based on sparkfun's example:

https://docs.sparkfun.com/SparkFun_Thing_Plus_ESP32-S3/arduino_example/#thing-plus-esp32-s3-github-repository

This guide uses the ESP32-S3 DEV Board and then redefines pins to adapt. I used the ESP32-S3 DEV Board as a base and made the changes in the boards.txt and the pins_arduino.h file to match the guide'options and the board's pin layout.

## Tests scenarios
I have tested my Pull Request with sparkfun's examples on Arduino-esp32 core v3.1.0 with ESP32 and ESP32-S3 Board.

## Related links
https://github.com/sparkfun/SparkFun_Thing_Plus_ESP32-S3
https://docs.sparkfun.com/SparkFun_Thing_Plus_ESP32-S3/arduino_example/#thing-plus-esp32-s3-github-repository

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -6904,6 +6904,266 @@ sparkfun_esp32s2_thing_plus.menu.EraseFlash.none.upload.erase_cmd=
...
+sparkfun_esp32s3_thing_plus.menu.FlashSize.4M=4MB (32Mb)
+sparkfun_esp32s3_thing_plus.menu.FlashSize.4M.build.flash_size=4MB
+sparkfun_esp32s3_thing_plus.menu.FlashSize.8M=8MB (64Mb)
+sparkfun_esp32s3_thing_plus.menu.FlashSize.8M.build.flash_size=8MB
+sparkfun_esp32s3_thing_plus.menu.FlashSize.8M.build.partitions=default_8MB
+sparkfun_esp32s3_thing_plus.menu.FlashSize.16M=16MB (128Mb)
+sparkfun_esp32s3_thing_plus.menu.FlashSize.16M.build.flash_size=16MB
+sparkfun_esp32s3_thing_plus.menu.FlashSize.32M=32MB (256Mb)
+sparkfun_esp32s3_thing_plus.menu.FlashSize.32M.build.flash_size=32MB
+sparkfun_esp32s3_thing_plus.menu.FlashSize.32M.build.partitions=app5M_fat24M_32MB
```
Comment (P-R-O-C-H-Y) [inline boards.txt:7007]: > This version of the ESP32-S3 has 4MB Flash memory and 2MB of dynamic/PSRAM

As the board uses the ESP32-S3-MINI-1-N4R2 with 4MB flash (QUAD) and 2MB PSRAM (QUAD), those options can be removed so users are not able to select wrong configuration.

```suggestion
```

File: boards.txt [hunk 2]
```diff
@@ -6904,6 +6904,266 @@ sparkfun_esp32s2_thing_plus.menu.EraseFlash.none.upload.erase_cmd=
...
+sparkfun_esp32s3_thing_plus.menu.UploadMode.cdc.upload.use_1200bps_touch=true
+sparkfun_esp32s3_thing_plus.menu.UploadMode.cdc.upload.wait_for_upload_port=true
+
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.default=Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS)
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.default.build.partitions=default
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.defaultffat=Default 4MB with ffat (1.2MB APP/1.5MB FATFS)
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.defaultffat.build.partitions=default_ffat
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.default_8MB=8M with spiffs (3MB APP/1.5MB SPIFFS)
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.default_8MB.build.partitions=default_8MB
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.default_8MB.upload.maximum_size=3342336
```
Comment (P-R-O-C-H-Y) [inline boards.txt:7058]: Not applicable for 4MB flash.
```suggestion
```

File: boards.txt [hunk 3]
```diff
@@ -6904,6 +6904,266 @@ sparkfun_esp32s2_thing_plus.menu.EraseFlash.none.upload.erase_cmd=
...
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.huge_app.upload.maximum_size=3145728
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.min_spiffs=Minimal SPIFFS (1.9MB APP with OTA/190KB SPIFFS)
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.min_spiffs.build.partitions=min_spiffs
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.min_spiffs.upload.maximum_size=1966080
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.fatflash=16M Flash (2MB APP/12.5MB FATFS)
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.fatflash.build.partitions=ffat
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.fatflash.upload.maximum_size=2097152
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.app3M_fat9M_16MB=16M Flash (3MB APP/9.9MB FATFS)
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.app3M_fat9M_16MB.build.partitions=app3M_fat9M_16MB
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.app3M_fat9M_16MB.upload.maximum_size=3145728
```
Comment (P-R-O-C-H-Y) [inline boards.txt:7087]: Same here.

```suggestion
```

File: boards.txt [hunk 4]
```diff
@@ -6904,6 +6904,266 @@ sparkfun_esp32s2_thing_plus.menu.EraseFlash.none.upload.erase_cmd=
...
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.app5M_fat24M_32MB=32M Flash (4.8MB APP/22MB FATFS)
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.app5M_fat24M_32MB.build.partitions=large_fat_32MB
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.app5M_fat24M_32MB.upload.maximum_size=4718592
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.app5M_little24M_32MB=32M Flash (4.8MB APP/22MB LittleFS)
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.app5M_little24M_32MB.build.partitions=large_littlefs_32MB
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.app5M_little24M_32MB.upload.maximum_size=4718592
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.esp_sr_16=ESP SR 16M (3MB APP/7MB SPIFFS/2.9MB MODEL)
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.esp_sr_16.upload.maximum_size=3145728
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.esp_sr_16.upload.extra_flags=0xD10000 {build.path}/srmodels.bin
+sparkfun_esp32s3_thing_plus.menu.PartitionScheme.esp_sr_16.build.partitions=esp_sr_16
```
Comment (P-R-O-C-H-Y) [inline boards.txt:7106]: Same here.

```suggestion
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@Vigeant PTAL

File: boards.txt (hunk 4 above)
Comment (Vigeant) [reply boards.txt]: made changes in last commit. Let me know if I should have created a branch before doing the pull request.

File: boards.txt (hunk 3 above)
Comment (Vigeant) [reply boards.txt]: done

File: boards.txt (hunk 2 above)
Comment (Vigeant) [reply boards.txt]: done

File: boards.txt (hunk 1 above)
Comment (Vigeant) [reply boards.txt]: done

**Review by P-R-O-C-H-Y — APPROVED** [review]

File: boards.txt (hunk 4 above)
Comment (P-R-O-C-H-Y) [reply boards.txt]: No it's fine like it is :) we will merge soon. Thanks for the fixes.

Comment (P-R-O-C-H-Y) [reply boards.txt]: But yeah, maybe next time create a branch with proper naming for the derived "changes" ;)

**Comment by Vigeant** [comment]
hey @P-R-O-C-H-Y how are you setup to use the latest master for your arduino IDE or PlatformIO setup?

**Pull request merged**